hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/constraints/ZookeeperWorkloadConstraintsPathChildrenCache.java

```diff
@@ -0,0 +1,42 @@
+package pl.allegro.tech.hermes.consumers.supervisor.workload.constraints;
+
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.recipes.cache.ChildData;
+import org.apache.curator.framework.recipes.cache.PathChildrenCache;
+import org.apache.curator.framework.recipes.cache.PathChildrenCacheEvent;
+import org.apache.curator.framework.recipes.cache.PathChildrenCacheListener;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+class ZookeeperWorkloadConstraintsPathChildrenCache extends PathChildrenCache implements PathChildrenCacheListener {
+
+    private final Map<String, ChildData> cache = new ConcurrentHashMap<>();
```

Comment:
I'm not sure if we need this. ` PathChildrenCache` internally stores a simple cache - see `List<ChildData> getCurrentData()`.